graylog2-web-interface/src/components/cluster/GraylogClusterOverview.jsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import Reflux from 'reflux';
+import { Row, Col } from 'react-bootstrap';
+
+import { Spinner } from 'components/common';
+
+import StoreProvider from 'injection/StoreProvider';
+const NodesStore = StoreProvider.getStore('Nodes');
+
+const GraylogClusterOverview = React.createClass({
+  mixins: [Reflux.listenTo(NodesStore, '_onNodesChange', '_onNodesChange')],
+
+  getInitialState() {
+    return {
+      clusterId: undefined,
+      nodeCount: 0,
+    };
+  },
+
+  _onNodesChange() {
+    this.setState({ clusterId: NodesStore.getClusterId(), nodeCount: NodesStore.getNodeCount() });
```

Comment:
Don't know why I didn't do that from the beginning. Using `listenTo` was actually problematic, as it could try to get some value from the store before the state was initialised.